config/shared/errors/errors.go

```diff
@@ -68,6 +68,8 @@ var (
 	ErrZeroesWithShouldNotExist  = errors.New("shouldExist is false for a partition and other partition(s) has start or size 0")
 	ErrNeedLabelOrNumber         = errors.New("a partition number >= 1 or a label must be specified")
 	ErrDuplicateLabels           = errors.New("cannot use the same partition label twice")
+	ErrInvalidProxy              = errors.New("proxies must be http(s)")
+	ErrInsecureProxy             = errors.New("http proxy specified for https proxy")
```

Comment:
Minor nit: I'd suggest to reword as "insecure plaintext HTTP proxy specified for HTTPS resources" for clarity.